pkg/ingester/flush.go

```diff
@@ -238,8 +247,11 @@ func (i *Ingester) flushUserSeries(flushQueueIndex int, userID string, fp model.
 		return nil
 	}
 
-	// shouldFlushSeries() has told us we have at least one chunk
-	chunks := series.chunkDescs
+	// shouldFlushSeries() has told us we have at least one chunk.
+	// Make a copy of chunks descriptors slice, to avoid possible issues related to removing (and niling) elements from chunkDesc.
+	// This can happen if first chunk is already flushed -- removeFlushedChunks may set such chunk to nil.
+	// Since elements in the slice are pointers, we can still safely update chunk descriptors after the copy.
+	chunks := append([]*desc(nil), series.chunkDescs...)
```

Comment:
A `make([]*desc, 0, len(series.chunkDescs)` + `copy()`  may me more efficient?

Reply:
I don't think sightly complicating code to be more efficient here will get us much... there is typically only one or two chunks in the slice. And I assume (without checking) that compiler is smart enough to optimize for this case anyway.